File: shared/types/category.ts
export interface Category {
  id: string;
  name: string;
  icon: string | null;
  sortOrder: number;
  isArchived: boolean;
}

export interface CreateCategoryRequest {
  name: string;
}

export interface UpdateCategoryRequest {
  name?: string;
  isArchived?: boolean;
}

File: backend/src/HowAreMyFinances.Api/Models/Category.cs
namespace HowAreMyFinances.Api.Models;

public sealed record Category(
    Guid Id,
    Guid UserId,
    string Name,
    string? Icon,
    int SortOrder,
    bool IsArchived,
    DateTime CreatedAt
);

public sealed record CreateCategoryRequest(string Name);

public sealed record UpdateCategoryRequest(string? Name, bool? IsArchived);
